target_chains/solana/programs/pyth-solana-receiver/src/lib.rs

```diff
@@ -507,6 +505,18 @@ fn post_twap_update_from_vaas<'info>(
 }
 
 fn calculate_twap(start_msg: &TwapMessage, end_msg: &TwapMessage) -> Result<(i64, u64, u32)> {
+    // Validate feed ids match
+    require!(
+        start_msg.feed_id == end_msg.feed_id,
+        ReceiverError::FeedIdMismatch
+    );
+
+    // Validate exponents match
+    require!(
+        start_msg.exponent == end_msg.exponent,
```

Comment:
almost feel like these checks shouldn't be in `calculate_twap` but in `post_twap_update`.
It should be immediately obvious reading the code that the assignment `twap.feed_id = start_msg.feed_id` is fine because we have that check so I recommend putting them closer.

Reply:
or even something like twap.feed_id = check_equal_and_return(start_msg.feed_id, end_msg.feed_id)

Reply:
agreed yeah, i had all the checks in `calculate_twap` to make it easy to unit test all the validations. i'll think i'll go the route of moving the validations out into `validate_twap_messages(start, end)` or similar so it remains easy to test and obvious in the context of assignments like `twap.feed_id = start_msg.feed_id` that checks have been done.